### ISSUE TITLE
Bugcheck during native program driver unload / NMR

### ISSUE THREAD

**Issue by mtfriesen** [microsoft/ebpf-for-windows#2130]
### Describe the bug

A bugcheck occurred when a native eBPF program was being unloaded.

### OS information

`17763.1.amd64fre.rs5_release.180914-1434`

### Steps taken to reproduce bug

Ran XDP stress test in the XDP project's CI/CD pipeline.

### Expected behavior

eBPF programs and drivers shouldn't cause bugchecks.

### Actual outcome

The system bugchecked.

### Additional details

```
nt!KeBugCheckEx+0x0 [minkernel\ntos\ke\amd64\procstat.asm @ 140]
nt!KiBugCheckDispatch+0x69 [minkernel\ntos\ke\amd64\trap.asm @ 3595]
nt!KiPageFault+0x44e [minkernel\ntos\ke\amd64\trap.asm @ 1560]
nt!KiAcquireSpinLockInstrumented+0x7f [minkernel\ntos\ke\spinlock.c @ 310]
nt!KeAcquireSpinLockRaiseToDpc+0x1ce3ca [minkernel\ntos\ke\spinlock.c @ 525]
verifierext!KeAcquireSpinLockRaiseToDpc_wrapper+0x65 [minkernel\verifier\xdv\rules\group_x64.c @ 6138]
nt!ViKeAcquireSpinLockRaiseToDpcCommon+0x3c [minkernel\ntos\verifier\vfke.c @ 1536]
nt!VerifierKeAcquireSpinLockRaiseToDpc+0x12 [minkernel\ntos\verifier\vfke.c @ 1588]
ebpfcore!+0x10e79  [D:\a\ebpf-for-windows\ebpf-for-windows\libs\execution_context\ebpf_native.c @ 271] (00000001`40010dd0)   eBPFCore!ebpf_native_release_reference+0xa9
ebpfcore!+0xeeb0  [D:\a\ebpf-for-windows\ebpf-for-windows\libs\execution_context\ebpf_native.c @ 467] (00000001`4000eda0)   eBPFCore!_ebpf_native_provider_detach_client_callback+0x110
netio!NmrpNotifyBindingDetach+0x48 [minio\netio\nmr\binding.c @ 494]
netio!NmrpDetachList+0x21 [minio\netio\nmr\module.c @ 494]
netio!NmrpDeregisterModule+0xc4 [minio\netio\nmr\module.c @ 564]
netio!NmrDeregisterClient+0x16 [minio\netio\nmr\module.c @ 740]
pass!+0x1256  [D:\a\1\s\artifacts\bin\x64_Release\bpf\pass_km\pass_driver.c @ 119] (00000001`40001240)   pass!DriverUnload+0x16
nt!IopLoadUnloadDriver+0x8d [minkernel\ntos\io\iomgr\internal.c @ 5250]
nt!ExpWorkerThread+0x16a [minkernel\ntos\ex\worker.c @ 4303]
nt!PspSystemThreadStartup+0x55 [minkernel\ntos\ps\psexec.c @ 8813]
nt!KiStartSystemThread+0x1c [minkernel\ntos\ke\amd64\threadbg.asm @ 81]
```

**Comment by mtfriesen** [comment]
The actual crash dump is on an internal system and cannot be exported.

**Comment by saxena-anurag** [comment]
This is fixed by #2355. Resolving this as duplicate of #2326
